### PR TITLE
Fix ggplotly error with NULL axis labs

### PR DESCRIPTION
**Pull request by doorisajar**
Resolves https://github.com/ropensci/plotly/issues/1731

### CODE REVIEW

File: R/ggplotly.R [hunk 1]
```diff
@@ -814,7 +814,7 @@ gg2list <- function(p, width = NULL, height = NULL,
           bbox(axisTickText, axisObj$tickangle, axisObj$tickfont$size)[[type]] +
           bbox(axisTitleText, axisTitle$angle, unitConvert(axisTitle, "pixels", type))[[type]]
         
-        if (nchar(axisTitleText) > 0) {
+        if (!is.null(axisTitleText) && nchar(axisTitleText) > 0) {
```
Comment (cpsievert) [inline R/ggplotly.R]:
```suggestion
        if (nzchar(axisTitleText %||% "")) {
```

File: R/ggplotly.R [hunk 2]
```diff
@@ -836,7 +836,7 @@ gg2list <- function(p, width = NULL, height = NULL,
           }
           # facets have multiple axis objects, but only one title for the plot,
           # so we empty the titles and try to draw the title as an annotation
-          if (nchar(axisTitleText) > 0) {
+          if (!is.null(axisTitleText) && nchar(axisTitleText) > 0) {
```
Comment (cpsievert) [inline R/ggplotly.R]:
```suggestion
          if (nzchar(axisTitleText %||% "")) {
```

File: R/ggplotly.R (hunk 1 above)
Comment (doorisajar) [reply R/ggplotly.R]: Sure, let me just make sure that behaves as expected with the plot title too.

File: R/ggplotly.R (hunk 2 above)
Comment (doorisajar) [reply R/ggplotly.R]: Interestingly, this variant fails the test, even though it definitely returns `FALSE` when passed a `NULL` or `logical(0)` explicitly. Not sure what's different about how `axisTitleText` is being received. For now I'll get it passing the test.

Comment (doorisajar) [reply R/ggplotly.R]: Totally forgot about `rlang::is_empty` for dealing with this stuff. That seems to do the trick.

Comment (cpsievert) [reply R/ggplotly.R]: plotly currently doesn't have an rlang dependency, so I'd rather use `length(x) == 0` than `rlang::is_empty()`

**Comment by cpsievert** [comment]
What happened in 3d99fad? Can you fix the diff on R/ggplotly.R?

**Comment by doorisajar** [comment]
> 
> 
> What happened in [3d99fad](https://github.com/ropensci/plotly/commit/3d99fad9876203d48a113f76b1638af59119e95b)? Can you fix the diff on R/ggplotly.R?

Looks like there's no line endings specification in the Rproj file, so when I switched from editing things in the GitHub interface to editing them in RStudio on Windows, the line endings got changed and I didn't notice. Just attempted to revert that in de198b4...

File: R/ggplotly.R (hunk 2 above)
Comment (doorisajar) [reply R/ggplotly.R]: `rlang` was already in the imports for `eval_tidy`: https://github.com/ropensci/plotly/blob/master/DESCRIPTION#L43

**Comment by cpsievert** [comment]
I'd still prefer to use `length()` directly over `rlang::is_empty()` as I more regularly use the former

**Comment by doorisajar** [comment]
> I'd still prefer to use `length()` directly over `rlang::is_empty()` as I more regularly use the former

In case you didn't see the commit, I did make this change.

**Comment by cpsievert** [comment]
Thanks for contributing this fix! It had some overlap with #1773 so we incorporated it there

**Pull request closed without merging**